libraries/protocol/include/graphene/protocol/config.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2015 Cryptonomex, Inc., and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+
+#define GRAPHENE_SYMBOL "BTS"
+#define GRAPHENE_ADDRESS_PREFIX "BTS"
+
+#define GRAPHENE_BLOCKCHAIN_PRECISION        uint64_t( 100000 )
+#define GRAPHENE_BLOCKCHAIN_PRECISION_DIGITS 5
+
+#define GRAPHENE_MIN_ACCOUNT_NAME_LENGTH 1
+#define GRAPHENE_MAX_ACCOUNT_NAME_LENGTH 63
+
+#define GRAPHENE_MIN_ASSET_SYMBOL_LENGTH 3
+#define GRAPHENE_MAX_ASSET_SYMBOL_LENGTH 16
+
+#define GRAPHENE_MAX_SHARE_SUPPLY int64_t(1000000000000000ll)
+
+#define GRAPHENE_MAX_WORKER_NAME_LENGTH                       63
+#define GRAPHENE_MAX_URL_LENGTH                               127
+
+#define GRAPHENE_MAX_SIG_CHECK_DEPTH 2
+
+#define GRAPHENE_IRREVERSIBLE_THRESHOLD                      (70 * GRAPHENE_1_PERCENT)
```

Comment:
IMHO this doesn't belong to protocol.

Reply:
It determines fork validity... I think it would be interesting to a third party implementation. Otherwise they would have to replicate this number in their own project, or risk their implementations following a fork that the reference implementation won't follow.

Reply:
The line between protocol and implementation is somewhat blurred. We have lots of consensus logic in the implementation, and all of it determines block validity.

I lean a bit towards @abitmore in this case. The threshold is only helpful for the local node, in that it defines any fork that goes back beyond LIB to be invalid. But an independent observer given two forks cannot decide which one is "real". Therefore the value is irrelevant wrt the protocol.